### ISSUE TITLE
Can't catch an exception on sigma's read method

### ISSUE THREAD

**Issue by tinchodipalma** [dunnock/react-sigma#29]
Any kind of exception that sigma throws as result of `addEdge` and `addNode` while is reading graph data can't be caught.

```
  constructor(props: Props) {
    super(props);
    this.state = {renderer:false}
    let settings = this.props.settings ? this.props.settings : {}
    this.sigma = new sigma({settings})
    this.initRenderer = this.initRenderer.bind(this)
    Sigma.bindHandlers(this.props, this.sigma)
    if(this.props.graph)
      this.sigma.graph.read(this.props.graph)
  }
```

What about adding a try/catch for the `read` method?

Regards,
Martin

**Comment by dunnock** [comment]
Spot on! Would you want to make a PR? Thanks

6 июн. 2017 г. 22:31 пользователь "Martin Di Palma" <
notifications@github.com> написал:

Any kind of exception that sigma throws as result of addEdge and addNode
while is reading graph data can't be caught.

  constructor(props: Props) {
    super(props);
    this.state = {renderer:false}
    let settings = this.props.settings ? this.props.settings : {}
    this.sigma = new sigma({settings})
    this.initRenderer = this.initRenderer.bind(this)
    Sigma.bindHandlers(this.props, this.sigma)
    if(this.props.graph)
      this.sigma.graph.read(this.props.graph)
  }

What about adding a try/catch for the read method?

Regards,
Martin

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/dunnock/react-sigma/issues/29>, or mute the thread

**Comment by dunnock** [comment]
Perfect, thanks @tinchodipalma ! I've only removed check for typeof onSigmaException, it is already checked by react propTypes transpiled from flow types by `flow-react-proptypes`, so only check that is defined is necessary.

**Comment by tinchodipalma** [comment]
Thanks @dunnock. Is it possible to publish a new minor/patch version?

**Comment by dunnock** [comment]
Published to npm react-sigma@1.2.19
